src/textures/mlx_xpm42.c

```diff
@@ -206,3 +206,28 @@ void mlx_delete_xpm42(xpm_t* xpm)
 	free(xpm->texture.pixels);
 	free(xpm);
 }
+
+char *convert_path(char* path)
+{
+    // Allocate memory for the new string
+    char *new_str = malloc(sizeof(char) * (strlen(path) + 3));
+    if (new_str == NULL) 
+	{
+        printf("Error: malloc failed\n");
+        return NULL;
+    }
+    // Copy the path to the new string and add "42"
+    strcpy(new_str, path);
+    strcat(new_str, "42");
+
+    // Build the command string to run the Python program
+    char command[256];
+    snprintf(command, sizeof(command), "python3 MLX42/tools/xpm3_conv.py \"%s\"", path);
+
```

Comment:
you can add a check for the script being not found for more clarity
```c
if (access(path, R_OK) == -1) {
    fputs("script not found error\n", stderr);
    return NULL;
}

Reply:
Not sure if thats a good idea, it's not really portable. You should be able to just compile your code and statically link the lib and thats it.

If you have to drag around python scripts to make it all work, it feels a bit silly.

Reply:
What if somehow they deleted the script I think it will be good to have a good message specifying what exactly went wrong instead of digging in the source code.

Reply:
Instead of having that problem to begin with, converting your png or jpeg straight to the right format is more logical, at least in my opinion, than having to rely on a python script at runtime.